### PR TITLE
easteregg/add-loading-popup

### PR DESCRIPTION
**Pull request by easteregg**

### CODE REVIEW

File: src/javascript/app/pages/trade/digit_trade.js [hunk 1]
```diff
@@ -38,12 +41,20 @@ const DigitDisplay = (() => {
...
                     .append($('<strong />', { class: 'gr-6', text: localize('Spot Time (GMT)') }))))
             .append($('<div />', { class: 'digit-ticker invisible', id: 'digit_ticker_container' }));
+        $digit_table = $('#table_digits');
+        $digit_table
+            .append($('<div />', { class: 'barspinner dark' })
+                .append($('<div />', { class: 'rect1' }))
+                .append($('<div />', { class: 'rect2' }))
+                .append($('<div />', { class: 'rect3' }))
+                .append($('<div />', { class: 'rect4' }))
+                .append($('<div />', { class: 'rect5' })));
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: is it worth moving the adding of a spinner to a utility function and passing `$digit_table` to it?

File: src/javascript/app/pages/trade/digit_trade.js [hunk 2]
```diff
@@ -101,7 +112,8 @@ const DigitDisplay = (() => {
         if (getPropertyValue(response, ['tick', 'id']) && document.getElementById('sell_content_wrapper')) {
             ViewPopupUI.storeSubscriptionID(response.tick.id);
         }
-
+        $digit_table.find('.barspinner').remove();
+        // Hide loading component
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: why is the comment below the line the does the action? 😅

Comment (easteregg) [reply src/javascript/app/pages/trade/digit_trade.js]: :joy: totally my bad.

File: src/javascript/app/pages/trade/digit_trade.js (hunk 1 above)
Comment (easteregg) [reply src/javascript/app/pages/trade/digit_trade.js]: Yeah, move it to it's own function.

File: src/javascript/app/pages/trade/digit_trade.js [hunk 3]
```diff
@@ -29,7 +30,20 @@ const DigitDisplay = (() => {
         }
     };
 
+    function setLoadingSpinner(content_id) {
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: you're expecting content_id, but on line 61 you're not passing any parameters.

File: src/javascript/app/pages/trade/digit_trade.js [hunk 4]
```diff
@@ -29,7 +30,20 @@ const DigitDisplay = (() => {
...
+        $digit_table = $(`#${content_id}`);
+        $digit_table
+            .append($('<div />', { class: 'barspinner dark' })
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: this isn't really what I meant. I meant moving this function to utility file etc so all pages can call it if needed. if we're still only specifically using it for digit table there's not much point in adding a separate function for it.

Comment (easteregg) [reply src/javascript/app/pages/trade/digit_trade.js]: I see what you mean now. This is a great idea. I generalize it a bit and move it to a separate module. (with show/hide functionality).

**Pull request merged**